code.js

```diff
@@ -1,3 +1,30 @@
 function binarySearch(list, element) {
+    if(list.length<1) return -1;//when list = [], return -1
+    var start = 0;
+    var end = list.length-1;
+    var mid;
+
+    while(start <= end) {
+        //always recursive in while need reset mid, otherwise mid will always the same number
+        mid = Math.floor((start+end)/2);
+        if(list[mid]===element){
+            // check if duplicate values in list
+            while (mid > 0 && list[mid - 1] === element) {
```

Comment:
This while loop is checking for duplicate values to the left of the original position where the target element was found. It will return the first occurrence position for the target element. If it is not a repeated value array, the list[mid] element equals to target element, it will return the element's index of the original array.